aptos-move/block-executor/src/scheduler.rs

```diff
@@ -464,17 +526,18 @@ impl Scheduler {
         let mut status = self.txn_status[txn_idx].lock();
         if let TransactionStatus::Suspended(incarnation, dep_condvar) = &*status {
```

Comment:
same here, we can match, treat suspended as is, ignore halted, and have the default (_ =>, everything else) branch unreachable